### ISSUE TITLE
Proposal: indent NewExpression

### ISSUE THREAD

**Issue by dbellavista** [eslint/eslint#8006]
**Please describe what the rule should do:**
Indent code like `indent` `CallExpression` but usable with the `new` operator 

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
/*eslint indent: ["error", 2, {"NewExpression": {"arguments": "first"}}]*/
let o = new MyObject(1,
  2, 3); // <-- Warn, should be aligned with first
```

```js
/*eslint indent: ["error", 2, {"NewExpression": {"arguments": "first"}}]*/
let o = new MyObject(1,
                     2, 3); // <-- Correct
```

**Why should this rule be included in ESLint (instead of a plugin)?**

The `new` operator use is widespread, especially with ES6. There should be an option to specify the constructor parameter indentation, just like for CallExpression.

**Comment by vitorbal** [comment]
@dbellavista thank your for your issue. It looks like what you are proposing is actually to enhance the `indent` rule, and not to create a new rule. In that case, could you fill in the [rule change proposal](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md) template instead? Thanks!

**Comment by dbellavista** [comment]
Oh right sorry :) #8011
